artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -424,6 +453,10 @@ public void preAcknowledge(final Transaction tx, final MessageReference ref, fin
 
       MirrorController controllerInUse = getControllerInUse();
 
+      if (controllerInUse != null && !controllerInUse.isAllowACK()) {
```

Comment:
This comment relates to code just above this change, not the change it is against.

The trace log at the start of the method looks wrong, references postACKInternalMessage when thats a seperate method further up, this is actually preAcknowledge

Reply:
Ohh... I see now.. when you said "this comment" I thought you were referring to this:

https://github.com/apache/activemq-artemis/blob/bc7e4639e0a4665c106f336fb837c33d2cbcd361/artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java#L427


That log. was already there before, and I see it now.. I am changing it.


and I also made an improvement to the code there as well.